op-acceptance-tests/tests/interop/reorgs/l2_reorgs_after_l1_reorg_test.go

```diff
@@ -0,0 +1,139 @@
+
+func TestL2ReorgAfterL1Reorg(gt *testing.T) {
+	gt.Run("unsafe reorg", func(gt *testing.T) {
+		var crossSafeRef, localSafeRef, unsafeRef eth.BlockID
+		pre := func(t devtest.T, sys *presets.SimpleInterop) {
+			crossSafeRef = sys.Supervisor.L2HeadBlockID(sys.L2ChainA.ChainID(), types.CrossSafe)
+			localSafeRef = sys.Supervisor.L2HeadBlockID(sys.L2ChainA.ChainID(), types.LocalSafe)
+			unsafeRef = sys.Supervisor.L2HeadBlockID(sys.L2ChainA.ChainID(), types.LocalUnsafe)
+		}
+		post := func(t devtest.T, sys *presets.SimpleInterop) {
+			require.True(t, sys.L2ELA.IsCanonical(crossSafeRef), "Previous cross-safe block should still be canonical")
+			require.True(t, sys.L2ELA.IsCanonical(localSafeRef), "Previous local-safe block should still be canonical")
+			require.False(t, sys.L2ELA.IsCanonical(unsafeRef), "Previous unsafe block should have been reorged")
+		}
+		testL2ReorgAfterL1Reorg(gt, 3, pre, post)
+	})
+
+	gt.Run("local-safe and cross-safe reorgs", func(gt *testing.T) {
+		var crossSafeRef, localSafeRef, unsafeRef eth.BlockID
+		pre := func(t devtest.T, sys *presets.SimpleInterop) {
+			crossSafeRef = sys.Supervisor.L2HeadBlockID(sys.L2ChainA.ChainID(), types.CrossSafe)
+			localSafeRef = sys.Supervisor.L2HeadBlockID(sys.L2ChainA.ChainID(), types.LocalSafe)
+			unsafeRef = sys.Supervisor.L2HeadBlockID(sys.L2ChainA.ChainID(), types.LocalUnsafe)
+		}
+		post := func(t devtest.T, sys *presets.SimpleInterop) {
+			require.False(t, sys.L2ELA.IsCanonical(crossSafeRef), "Previous cross-safe block should have been reorged")
+			require.False(t, sys.L2ELA.IsCanonical(localSafeRef), "Previous local-safe block should have been reorged")
+			require.False(t, sys.L2ELA.IsCanonical(unsafeRef), "Previous unsafe block should have been reorged")
+		}
+		testL2ReorgAfterL1Reorg(gt, 10, pre, post)
+	})
+}
+
+// testL2ReorgAfterL1Reorg tests that the L2 chain reorgs after an L1 reorg, and takes n, number of blocks to reorg, as parameter
+// for unsafe reorgs - n must be at least >= confDepth, which is 2 in our test deployments
+// for cross-safe reorgs - n must be at least >= safe distance, which is 10 in our test deployments
+// pre- and post-checks are sanity checks to ensure that the blocks we expected to be reorged were indeed reorged or not
+func testL2ReorgAfterL1Reorg(gt *testing.T, n int, preChecks, postChecks checksFunc) {
+	t := devtest.SerialT(gt)
+	ctx := t.Ctx()
+
+	sys := presets.NewSimpleInterop(t)
+	ts := sys.TestSequencer.Escape().ControlAPI(sys.L1Network.ChainID())
+
+	cl := sys.L1Network.Escape().L1CLNode(match.FirstL1CL)
+
+	sys.L1Network.WaitForBlock()
+
+	sys.ControlPlane.FakePoSState(cl.ID(), stack.Stop)
+
+	// sequence a few L1 and L2 blocks
+	for range n + 1 {
+		sequenceL1Block(t, ts, common.Hash{})
+
+		sys.L2ChainA.WaitForBlock()
+		sys.L2ChainA.WaitForBlock()
+	}
+
+	// select a divergence block to reorg from
+	var divergence eth.BlockInfo
+	{
+		tip, err := sys.L1EL.Escape().EthClient().InfoByLabel(ctx, "latest")
```

Comment:
```suggestion
		tip, err := sys.L1EL.Escape().EthClient().InfoByLabel(ctx, eth.Unsafe)
```
But would be nice to wrap this as a DSL method